repos/spack_repo/builtin/packages/py_sentry_sdk/package.py

```diff
@@ -32,20 +33,24 @@ class PySentrySdk(PythonPackage):
     variant("chalice", default=False, description="Builts with chalice")
     variant("httpx", default=False, when="@1.5.5:", description="Builts with httpx")
 
+    depends_on("python@3.6:", type=("build", "run"), when="@2:")
     depends_on("python@2.7,3.4:", type=("build", "run"))
```

Comment:
You probably want to add the appropriate condition here, e.g.


```suggestion
    depends_on("python@2.7,3.4:", type=("build", "run"), when="@:1")
```

Reply:
Same for the other dependencies whose versions have changed for the newer release(s).

Reply:
I'm sorry, I don't understand why this is necessary. Ideally, the exact upper version bound should be specified, which adds a lot of work. Usually they are not even documented. Why is the current solution not satisfactory?

Reply:
Being explicit when a dependency doesn't apply should reduce the numbers of dependencies that have to be considered during concretization.